lib/kitchen/driver/openstack.rb

```diff
@@ -316,9 +340,10 @@ def setup_ssh(server, state)
 
       def do_ssh_setup(state, config, server)
         info "Setting up SSH access for key <#{config[:public_key_path]}>"
+        password = config[:password].nil? ? server.password : config[:password]
         ssh = Fog::SSH.new(state[:hostname],
                            config[:username],
-                           password: server.password)
+                           password: password)
```

Comment:
This looks fine, though FYI could also be simplified to `password: config[:password] || server.password`.